config/jobs/kubernetes-csi/gen-jobs.sh

```diff
@@ -610,42 +610,42 @@ for deployment_suffix in "" "-test"; do
                 fi
                 actual="$(if [ "$kubernetes" = "master" ]; then echo latest; else echo "release-$kubernetes"; fi)"
                 cat >>"$base/csi-driver-host-path/csi-driver-host-path-config.yaml" <<EOF
-- interval: 6h
-  name: $(job_name "ci" "" "$tests" "$deployment$deployment_suffix" "$kubernetes")
-  decorate: true
-  extra_refs:
-  - org: kubernetes-csi
-    repo: csi-driver-host-path
-    base_ref: master
-  labels:
-    preset-service-account: "true"
-    preset-dind-enabled: "true"
-    preset-kind-volume-mounts: "true"
-  $(annotations "    " "ci" "" "$tests" "$deployment$deployment_suffix" "$kubernetes")
-  spec:
-    containers:
-    # We need this image because it has Docker in Docker and go.
-    - image: ${dind_image}
-      command:
-      - runner.sh
-      args:
-      - ./.prow.sh
-      env:
-      - name: CSI_PROW_KUBERNETES_VERSION
-        value: "$actual"
-      - name: CSI_SNAPSHOTTER_VERSION
-        value: $(snapshotter_version "$actual" "")
-      - name: CSI_PROW_BUILD_JOB
-        value: "false"
-      - name: CSI_PROW_DEPLOYMENT
-        value: "kubernetes-$deployment"
-      - name: CSI_PROW_DEPLOYMENT_SUFFIX
-        value: "$deployment_suffix"
-      - name: CSI_PROW_TESTS
-        value: "$(expand_tests "$tests")"
-      # docker-in-docker needs privileged mode
-      securityContext:
-        privileged: true
+  - interval: 6h
```

Comment:
Is it really necessary to change the indention of the periodic jobs? That is what makes the diff in the generated files so large and thus harder to review.

The other reason is that some previous change wasn't committed, see https://github.com/kubernetes/test-infra/pull/27166#issuecomment-1243250325

Reply:
unfortunately its necessary due to the dynamic resources injection... the issue is that periodics and presubmits were at different indentation levels and when the resources were added they would break for one or the other... unless there is some way to read the indentation level programatically ... This is a simpler solution...

Reply:
apologies for the large size tho

Reply:
So the validation is based on plain text checks instead of parsing the YAML?

That seems rather fragile and unexpected, because a perfectly fine job would get rejected. Okay:
```
periodics:
 - ...
```

Not okay:
```
periods:
- ...
```

Did I understand this correctly?

Reply:
Not validation - "dynamic resources injection", whethever that is. Do you have a pointer? But the same concern applies.

Reply:
maybe I explained it incorrectly... the script `gen-jobs.sh` builds the jobs using bash and plain text (`<<EOF EOF`)... here the `presubmits:` jobs have a `kubernetes-csi/$repo:` tag followed(in the next indentation level), but the periodics: has no such tag and has jobs in the second indentation level instead of the 3rd... when the resources are added (plain text append) they are added with the indentation level so it will either break for one or the other... 

the validation is done with the checkconfig tool that I do believe parses YAML

Reply:
I agree having the script essentially echoing YAML as plain text is pretty fragile thats why it was giving the resources indentation bug

Reply:
> when the resources are added (plain text append based on job) they are added with the indentation level so it will either break for one or the other...

Now I understand what you meant. But forcing the surrounding text to have a certain indention for `resources_for_kubernetes` to work correctly seems backwards to me. A better approach would be to add another parameter to it which contains the spaces that need to be inserted before `resources:` and all following lines.

Reply:
ah... that makes sense... Ill give that a go